tests/unittests/test_error_handling.py

```diff
@@ -1,23 +1,49 @@
 import unittest
-from unittest.mock import Mock
+from unittest import mock
 from urllib.error import URLError
 import tap_shopify
+from tap_shopify.streams.transactions import Transactions
 from tap_shopify.context import Context
 
 
 class TestShopifyConnectionResetErrorHandling(unittest.TestCase):
 
-    def test_check_access_handle_timeout_error(self):
+    @mock.patch("time.sleep")
+    @mock.patch("pyactiveresource.activeresource.ActiveResource.find")
+    def test_check_access_handle_timeout_error(self, mocked_find, mocked_sleep):
         '''
         Test retry handling of URLError
         '''
 
-        stream = Context.stream_objects['orders']()
-        stream.replication_object = Mock()
-        stream.replication_object.find = Mock()
-        stream.replication_object.find.side_effect = URLError('<urlopen error [Errno 104] Connection reset by peer>')
+        # mock 'find' and raise URLError
+        mocked_find.side_effect = URLError('<urlopen error [Errno 104] Connection reset by peer>')
 
-        with self.assertRaises(URLError):
+        #initialize class
+        stream = Context.stream_objects['orders']()
+        try:
+            #function call
             stream.call_api({})
+        except URLError:
+            pass
+
+        # verify we backoff 5 times
+        self.assertEqual(mocked_find.call_count, 5)
+
+    @mock.patch("time.sleep")
+    @mock.patch("pyactiveresource.activeresource.ActiveResource.find")
+    def test_check_access_handle_connection_reset_error(self, mocked_find, mocked_sleep):
+        '''
+        Test retry handling of ConnectionResetError
+        '''
+        # mock 'find' and raise IncompleteRead error
+        mocked_find.side_effect = ConnectionResetError
+        # initialize class
+        locations = Transactions()
+        try:
+            # function call
+            locations.replication_object.find()
+        except ConnectionResetError:
+            pass
 
-        self.assertEqual(stream.replication_object.find.call_count, 5)
+        # verify we backoff 5 times
+        self.assertEquals(mocked_find.call_count, 5)
```

Comment:
Instead of hard coding the call_count to be 5, we can import `MAX_RETRIES` from `from tap_shopify.streams.base` and use it here for assertion

Reply:
I think, we should not use `MAX_RETRIES` from `tap_shopify.streams.base` in assertion because unittests would not catch any deviation of `MAX_RETRIES`from the expectations.